packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -494,83 +499,150 @@ export class FungibleTokenTransferController implements ReactiveController {
     this.host.requestUpdate();
   };
 
-  private buildTransactions(): void {
+  private canBuildTransactions(): boolean {
     if (
       !this.sourceNetwork ||
       !this.destinationNetwork ||
       !this.resourceAmount ||
       !this.selectedResource ||
-      !this.destinationAddress
+      !this.destinationAddress ||
+      this.isBuildingTransactions
     ) {
-      return;
+      return false;
     }
+
     switch (this.sourceNetwork.type) {
+      case Network.EVM: {
+        const address = this.walletContext.value?.evmWallet?.address;
+        const provider = this.walletContext.value?.evmWallet?.provider;
+        const providerChainId =
+          this.walletContext.value?.evmWallet?.providerChainId;
+        if (!address || !provider || !providerChainId) return false;
+        return true;
+      }
+      case Network.SUBSTRATE: {
+        const substrateProvider = this.sourceSubstrateProvider;
+        const substrateAddress =
+          this.walletContext.value?.substrateWallet?.signerAddress;
+        if (!substrateProvider || !substrateAddress) return false;
+        return true;
+      }
+    }
+  }
+
+  private async buildTransactions(): Promise<void> {
+    const isAbleToBuildTransactions = this.canBuildTransactions();
+
+    if (!isAbleToBuildTransactions) {
+      this.estimatedGas = undefined;
+      this.resetFee();
+      return;
+    }
+
+    switch (this.sourceNetwork!.type) {
       case Network.EVM:
         {
-          void this.buildEvmTransactions();
+          const address = this.walletContext.value?.evmWallet?.address;
+          const provider = this.walletContext.value?.evmWallet?.provider;
+          const providerChainId =
+            this.walletContext.value?.evmWallet?.providerChainId;
```

Comment:
maybe it'd be cleaner if you just pass `evmWallet` as one param, instead of 3 (and unwrap in function)

Reply:
you'd need to change function signature, but still seems a bit better than unwrapping every time manually?